operations/configuration_management/Pre_Boot_NCN_Image_Configuration.md

```diff
@@ -0,0 +1,138 @@
+
+   If modifying anything at the kernel-level, there is no need to create a new `initrd`. In
+   that case, simply use the existing `initrd`.
+
+   **NOTE:** If desired, the default SSH *host* keys can be removed with a CFS Ansible task:
+
+   ```console
+   - name: Remove SSH host keys
+     shell: rm -f /etc/ssh/*key*
+     warn: false
+   ```
+
+   New host keys will be generated by `sshd` when the NCN boots for the first time.
+
+1. [Create a CFS Configuration](Create_a_CFS_Configuration.md)
+
+1. [Create an Image Customization CFS Session](Create_an_Image_Customization_CFS_Session.md)
+
+1. (`ncn#`) Download the Resultant NCN Artifacts
+
+    **NOTE:** `$IMS_RESULTANT_IMAGE_ID` is the `result_id` returned in the output of the last command
+    in the previous step:
+
+    ```bash
+    cray cfs sessions describe example --format json | jq .status.artifacts
+    ```
+
+    **NOTE:** If `create-ims-initrd.sh` was not run in the CFS session, do *not* download the `initrd` and `kernel`. Only download the SquashFS file.
+
+    ```bash
+    cray artifacts get boot-images $IMS_RESULTANT_IMAGE_ID/rootfs kubernetes-$ARTIFACT_VERSION-1.squashfs
+
+    ncn# cray artifacts get boot-images $IMS_RESULTANT_IMAGE_ID/initrd initrd.img-$ARTIFACT_VERSION-1.xz
+
+    ncn# cray artifacts get boot-images $IMS_RESULTANT_IMAGE_ID/kernel 5.3.18-150300.59.43-default-$ARTIFACT_VERSION-1.kernel
+    ```
+
+1. (`ncn#`) Upload NCN boot artifacts into S3
+
+    This steps assumes that the `docs-csm` RPM is installed.
+
+    **NOTE:** If `create-ims-initrd.sh` was not run in the CFS session, use the same `initrd` and `kernel` that were retrieved in step 1.
+
+    ```bash
+    /usr/share/doc/csm/scripts/ceph-upload-file-public-read.py --bucket-name ncn-images --key-name "k8s/$ARTIFACT_VERSION-1/filesystem.squashfs" --file-name kubernetes-$ARTIFACT_VERSION-1.squashfs
+
+    /usr/share/doc/csm/scripts/ceph-upload-file-public-read.py --bucket-name ncn-images --key-name "k8s/$ARTIFACT_VERSION-1/initrd" --file-name initrd.img-$ARTIFACT_VERSION-1.xz
+
+    /usr/share/doc/csm/scripts/ceph-upload-file-public-read.py --bucket-name ncn-images --key-name "k8s/$ARTIFACT_VERSION-1/kernel" --file-name $ARTIFACT_VERSION-1.kernel
+    ```
+
+1. (`ncn#`) Update NCN Boot Parameters
+
+    Get the existing `metal.server` setting for the xname of the node of interest:
+
+    ```bash
+    XNAME=<your-xname>
+    METAL_SERVER=$(cray bss bootparameters list --hosts $XNAME --format json | jq '.[] |."params"' \
+         | awk -F 'metal.server=' '{print $2}' \
+         | awk -F ' ' '{print $1}')
+    ```
+
+    Verify the variable was set correctly: `echo $METAL_SERVER`
+
+    Update the kernel, initrd, and metal server to point to the new artifacts.
+
+    ```bash
+    S3_ARTIFACT_PATH=ncn-images/k8s/$ARTIFACT_VERSION-1
+    NEW_METAL_SERVER=http://rgw-vip.nmn/$S3_ARTIFACT_PATH
+
+    PARAMS=$(cray bss bootparameters list --hosts $XNAME --format json | jq '.[] |."params"' | \
```

Comment:
Since we do not wipe, and we want to use the auto-wipe, we have to set `metal.no-wipe=0` at some point before rebuilding.

Reply:
fixed